### PR TITLE
LPS-185065 Do not use objectFieldSettings to relate readOnly property

### PR DESCRIPTION
**Pull request by danvb112**
Related Issue: https://issues.liferay.com/browse/LPS-185065

### CODE REVIEW

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/Layout/LayoutScreen/ModalAddObjectLayoutField.tsx [hunk 1]
```diff
@@ -174,9 +173,8 @@ export default function ModalAddObjectLayoutField({
 								</ClayLabel>
 
 								{Liferay.FeatureFlags['LPS-170122'] &&
-									(readOnlyField.value === 'true' ||
-										readOnlyField.value ===
-											'conditional') && (
+									(readOnlyField === 'true' ||
```
Comment (naraandrade) [inline modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/Layout/LayoutScreen/ModalAddObjectLayoutField.tsx:176]: Hi @danvb112 everything works fine in my tests. Could you just order this verification by alphabetical order? Like 
`readOnlyField === 'conditional' || readOnlyField === 'true'`. Thanks :) 

Please apply that change in the other places that occurs as well.

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectField/AddObjectField.tsx [hunk 2]
```diff
@@ -63,10 +63,20 @@ function ModalAddObjectField({
...
 		listTypeDefinitionExternalReferenceCode: '',
 		listTypeDefinitionId: 0,
+		readOnly: 'false',
+		readOnlyConditionExpression: '',
 		required: false,
 	};
 
 	const onSubmit = async (field: Partial<ObjectField>) => {
+		if (
+			field.businessType === 'Formula' ||
```
Comment (naraandrade) [inline modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectField/AddObjectField.tsx:73]: @danvb112 Similar to the comment above. Could you reorder it like
`field.businessType === 'Aggregation' || field.businessType === 'Formula'`? Thanks :)

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/index.d.ts [hunk 3]
```diff
@@ -154,6 +156,8 @@ interface ObjectDefinition {
 	titleObjectFieldName: string;
 }
 
+type ReadOnlyFieldValue = '' | 'true' | 'false' | 'conditional';
```
Comment (naraandrade) [inline modules/apps/object/object-web/src/main/resources/META-INF/resources/js/index.d.ts:159]: @danvb112 Please, order them in alphabetical order. Thanks :)

**Review by naraandrade — CHANGES_REQUESTED** [review]

**Comment by danvb112** [comment]
ci:test:sf

**Comment by danvb112** [comment]
ci:test:sf

**Comment by danvb112** [comment]
ci:test:sf

**Comment by danvb112** [comment]
ci:test:sf

**Comment by danvb112** [comment]
ci:test:sf

**Comment by danvb112** [comment]
ci:test:relevant

**Review by naraandrade — APPROVED** [review]

**Comment by danvb112** [comment]
Sent to: https://github.com/brianchandotcom/liferay-portal/pull/135752

**Pull request closed without merging**